tests/test_middleware_interface.py

```diff
@@ -228,17 +263,21 @@ def test_ingest_image_fails_missing_file(self):
         self.assertEqual(datasets, [])
 
     def test_run_pipeline(self):
-        # Can't run an actual pipeline because raw/calib/refcat/template data are all zeroed out.
-        with unittest.mock.patch('activator.middleware_interface.SimplePipelineExecutor') as mock_executor:
+        """Test that running the pipeline uses the correct arguments.
+        We can't run an actual pipeline because raw/calib/refcat/template data
+        are all zeroed out.
+        """
+        # Have to setup the data so that we can create the pipeline executor.
+        self.interface.prep_butler(self.next_visit)
+        filename = "fakeRawImage.fits"
+        filepath = os.path.join(self.input_data, filename)
+        data_id, file_data = fake_file_data(filepath,
+                                            self.butler.dimensions,
+                                            self.interface.instrument)
+        with unittest.mock.patch.object(self.interface.rawIngestTask, "extractMetadata") as mock:
+            mock.return_value = file_data
+            self.interface.ingest_image(filename)
+
+        with unittest.mock.patch("activator.middleware_interface.SimplePipelineExecutor.run") as run:
```

Comment:
`mock_run`? I'd like to keep straight which things are mocks and which are real.